src/precice/bindings/python_future/precice_future.pyx

```diff
@@ -182,47 +185,50 @@ cdef class Interface:
     def map_write_data_from (self, int from_mesh_id):
         self.thisptr.mapWriteDataFrom (from_mesh_id)
 
-    def write_block_vector_data (self, int data_id, value_indices, np.ndarray[np.double_t, ndim=1] values):
-        assert(values.size / self.get_dimensions() == value_indices.size)
+    def write_block_vector_data (self, int data_id, value_indices, values):
+        cdef np.ndarray[int, ndim=1] _value_indices = np.ascontiguousarray(value_indices, dtype=np.int32)
+        cdef np.ndarray[double, ndim=1] _values = np.ascontiguousarray(values, dtype=np.double)
+        assert(_values.size / self.get_dimensions() == value_indices.size)
         size = value_indices.size
-        cdef np.ndarray[int, ndim=1] value_indices_int = np.array(value_indices, dtype=np.int32)
-        self.thisptr.writeBlockVectorData (data_id, size, &value_indices_int[0], &values[0])
+        self.thisptr.writeBlockVectorData (data_id, size, &_value_indices[0], &_values[0])
 
-    def write_vector_data (self, int data_id, value_index, np.ndarray[np.double_t, ndim=1] value):
-        self.thisptr.writeVectorData (data_id, value_index, &value[0])
+    def write_vector_data (self, int data_id, value_index, value):
+        cdef np.ndarray[np.double_t, ndim=1] _value = np.ascontiguousarray(value, dtype=np.double)
+        self.thisptr.writeVectorData (data_id, value_index, &_value[0])
 
-    def write_block_scalar_data (self, int data_id, value_indices, np.ndarray[np.double_t, ndim=1] values):
+    def write_block_scalar_data (self, int data_id, value_indices, values):
+        cdef np.ndarray[int, ndim=1] _value_indices = np.ascontiguousarray(value_indices, dtype=np.int32)
+        cdef np.ndarray[double, ndim=1] _values = np.ascontiguousarray(values, dtype=np.double)
         assert(values.size == value_indices.size)
         size = value_indices.size
-        cdef np.ndarray[int, ndim=1] value_indices_int = np.array(value_indices, dtype=np.int32)
-        self.thisptr.writeBlockScalarData (data_id, size, &value_indices_int[0], &values[0])
+        self.thisptr.writeBlockScalarData (data_id, size, &_value_indices[0], &_values[0])
 
     def write_scalar_data (self, int data_id, int value_index, double value):
         self.thisptr.writeScalarData (data_id, value_index, value)
 
     def read_block_vector_data (self, int data_id, value_indices):
-        size = value_indices.size
-        cdef np.ndarray[int, ndim=1] value_indices_int = np.array(value_indices, dtype=np.int32)
-        cdef np.ndarray[np.double_t] values = np.empty(size * self.get_dimensions(), dtype=np.double)
-        self.thisptr.readBlockVectorData (data_id, size, &value_indices_int[0], &values[0])
-        return values
+        cdef np.ndarray[int, ndim=1] _value_indices = np.ascontiguousarray(value_indices, dtype=np.int32)
+        size = _value_indices.size
+        cdef np.ndarray[np.double_t, ndim=1] _values = np.empty(size * self.get_dimensions(), dtype=np.double)
```

Comment:
I agree. This would be a nice feature, but in my opinion it would also be a clear API change. How about opening an issue and improving this at a later point in time?